source/extensions/health_checkers/redis/utility.h

```diff
@@ -17,11 +18,19 @@ namespace {
 static const envoy::config::health_checker::redis::v2::Redis
 getRedisHealthCheckConfig(const envoy::config::core::v3alpha::HealthCheck& health_check_config,
                           ProtobufMessage::ValidationVisitor& validation_visitor) {
+
   ProtobufTypes::MessagePtr config =
       ProtobufTypes::MessagePtr{new envoy::config::health_checker::redis::v2::Redis()};
-  MessageUtil::jsonConvert(
-      health_check_config.custom_health_check().hidden_envoy_deprecated_config(),
-      validation_visitor, *config);
+  if (health_check_config.custom_health_check().has_typed_config()) {
+    Envoy::Config::Utility::translateOpaqueConfig(
+        health_check_config.custom_health_check().typed_config(),
```

Comment:
@mattklein123 looks like we found a bug in the process.

Reply:
Nice find. I think this can all be collapsed into a single call into `translateOpaqueConfig(..)` by passing `health_check_config.custom_health_check().hidden_envoy_deprecated_config()` as the second param? cc @HenryYYang